### PR TITLE
attempting to refactor and make sense of looper

### PR DESCRIPTION
**Pull request by MikeSilvis**
DO NOT MERGE THIS.

I have no clue what will happen as I never actually tested anything. Just trying to show off some of the things ruby can do.

NEVER use a `for` statement in ruby... And try as hard as you can to prevent a double nested loop... 

http://ruby-doc.org/core-2.0/Enumerable.html

Enumerable is so powerful. Checkout things like: select, detect, map ( i saw you use collect which is the same thing), inject, etc. 

This project from HA: http://tutorials.jumpstartlab.com/projects/sales_engine.html really helped me learn enumerable in ruby and all of the amazing things it can do.

here is my example: https://github.com/MikeSilvis/sales_engine 

But I would REALLY try to do it on your own and TDD it. I learned so much from this project.

### CODE REVIEW

**Comment by MikeSilvis** [comment]
While building the above project try to think about things you use in rails like: has_many, belongs_to etc and how you would write them.

**Comment by MikeSilvis** [comment]
see: https://github.com/MikeSilvis/ultimate_whip/blob/master/app/lib/image_scrapper.rb

I'm not sure its the best ruby but it should read like english. 

If you start at ImageScraper.new.find_all_images it reads like a story. 

Small methods make it easier to test, and easier to read, and easier to refactor. sales engine taught me that little methods (even if not reused) make for cleaner easier to understand code.

File: app/models/user.rb [hunk 1]
```diff
@@ -75,6 +71,10 @@ def update_with_password(params, *options)
     end
   end
 
+
+  # it appears you are constantly fighting devise to get rid of the required fields. Do you ever 
+  # let someone login not via facebook? If not at the top on line 11 you can comment out certain parts
+  # and it will give rid of these requirements. Otherwise this is a great way to override methods.
```
Comment (psilvis) [inline app/models/user.rb:77]: The original plan was to allow both; but as the project evolved we decided there was no value in traditional accounts (nobody to invite).

This override came compliments of railscasts

File: app/models/share.rb [hunk 2]
```diff
@@ -26,30 +22,39 @@ def embedly
...
-          if user.id == view.user_id
-            usr[:viewed] = 1
-          end
-        end
-        viewers << usr
-      end
-      viewers
-    else
-      []
+  alias :scrape :embedly # Not sure why you are defining a method that just calls another but you can do this
```
Comment (psilvis) [inline app/models/share.rb:25]: was a longer method at one time.  recently hacked in embedly() to replace the youtube-it & soundcloud gems.  Useful tip though thanks!

Comment (MikeSilvis) [inline app/models/share.rb:25]: yeah i saw you later used alias. But I wanted to show it off to you before i saw that.

**Pull request closed without merging**